core/templates/domain/topic/topic-object.model.ts

```diff
@@ -584,40 +584,47 @@ export class Topic {
   // Reassigns all values within this topic to match the existing
   // topic. This is performed as a deep copy such that none of the
   // internal, bindable objects are changed within this topic.
-  copyFromTopic(otherTopic: Topic): void {
-    this._id = otherTopic.getId();
-    this.setName(otherTopic.getName());
-    this.setAbbreviatedName(otherTopic.getAbbreviatedName());
-    this.setUrlFragment(otherTopic.getUrlFragment());
-    this.setThumbnailFilename(otherTopic.getThumbnailFilename());
-    this.setThumbnailBgColor(otherTopic.getThumbnailBgColor());
-    this.setDescription(otherTopic.getDescription());
-    this.setLanguageCode(otherTopic.getLanguageCode());
-    this.setPracticeTabIsDisplayed(otherTopic.getPracticeTabIsDisplayed());
-    this.setMetaTagContent(otherTopic.getMetaTagContent());
-    this.setPageTitleFragmentForWeb(otherTopic.getPageTitleFragmentForWeb());
-    this.setSkillSummariesForDiagnosticTest(
-      otherTopic.getSkillSummariesForDiagnosticTest());
-    this._version = otherTopic.getVersion();
-    this._nextSubtopicId = otherTopic.getNextSubtopicId();
-    this.clearAdditionalStoryReferences();
-    this.clearCanonicalStoryReferences();
-    this.clearUncategorizedSkills();
-    this.clearSubtopics();
-
-    this._canonicalStoryReferences = otherTopic.getCanonicalStoryReferences();
-    this._additionalStoryReferences =
+  copyFromTopic(otherTopic: Topic): Topic {
+    let id = otherTopic.getId();
+    let name = otherTopic.getName();
+    let abbreviatedName = otherTopic.getAbbreviatedName();
+    let urlFragment = otherTopic.getUrlFragment();
+    let thumbnailFilename = otherTopic.getThumbnailFilename();
+    let thumbnailBgColor = otherTopic.getThumbnailBgColor();
+    let description = otherTopic.getDescription();
+    let languageCode = otherTopic.getLanguageCode();
+    let practiceTabIsDisplayed = otherTopic.getPracticeTabIsDisplayed();
+    let metaTagContent = otherTopic.getMetaTagContent();
+    let pageTitleFragmentForWeb = otherTopic.getPageTitleFragmentForWeb();
+    let skillSummariesForDiagnosticTest =
+      otherTopic.getSkillSummariesForDiagnosticTest();
+    let version = otherTopic.getVersion();
+    let nextSubtopicId = otherTopic.getNextSubtopicId();
+    let topic = new Topic(
```

Comment:
Add a line break above this one, because what comes after this point is somewhat different from what came before.

Reply:
Done